pyccel/parser/syntactic.py

```diff
@@ -750,11 +750,14 @@ def fill_types(ls):
                 else:
                     types = fill_types(ls)
 
-                txt  = '#$ header ' + name
-                txt += '(' + ','.join(types) + ')'
+                txt  = '#$ header '
+                if len(self._context) > 1 and isinstance(self._context[-2], ast.ClassDef):
+                    txt += 'method '
+                txt += name + '(' + ','.join(types) + ')'
 
                 if results:
                     txt += ' results(' + ','.join(results) + ')'
+                print(txt)
```

Comment:
Is this necessary ?

Reply:
Oops. Good spot I was trying (unsuccessfully) to debug and forgot about it